File: src/main/java/de/saschahlusiak/freebloks/network/NET_REVOKE_PLAYER.java
package de.saschahlusiak.freebloks.network;

import java.io.ByteArrayOutputStream;

public class NET_REVOKE_PLAYER extends NET_HEADER {
	public int player; /* int 8 */

	public NET_REVOKE_PLAYER(int player) {
		super(MessageType.RevokePlayer, 1);
		this.player = player;
	}

	@Override
	void prepare(ByteArrayOutputStream bos) {
		super.prepare(bos);
		bos.write(player);
	}
}

File: src/main/java/de/saschahlusiak/freebloks/network/message/NetRevokePlayer.kt
package de.saschahlusiak.freebloks.network.message

import de.saschahlusiak.freebloks.network.*
import java.nio.ByteBuffer

data class NetRevokePlayer(val player: Int): Message(MessageType.RevokePlayer, 1) {
    init {
        assert(player in 0..3) { "Invalid player $player" }
    }

    override fun write(buffer: ByteBuffer) {
        super.write(buffer)
        buffer.put(player.toByte())
    }

    companion object {
        fun from(data: ByteBuffer): NetRevokePlayer {
            val player = data.get().toInt()
            return NetRevokePlayer(player)
        }
    }
}